File: GitToolBox/src/main/java/zielu/gittoolbox/completion/CompletionScopeProvider.java
package zielu.gittoolbox.completion;

import java.io.File;
import java.util.Collection;
import java.util.Collections;
import org.jetbrains.annotations.NotNull;

interface CompletionScopeProvider {
  CompletionScopeProvider EMPTY = Collections::emptyList;

  @NotNull
  Collection<File> getAffectedFiles();
}

File: src/main/kotlin/zielu/gittoolbox/completion/CompletionScopeProvider.kt
package zielu.gittoolbox.completion

import java.io.File

internal interface CompletionScopeProvider {
  fun getAffectedFiles(): Collection<File>

  companion object {
    val empty: CompletionScopeProvider = object : CompletionScopeProvider {
      override fun getAffectedFiles(): Collection<File> = listOf()
    }
  }
}
